### ISSUE TITLE
How about support GoStruct encode?

### ISSUE THREAD

**Issue by eyotang** [mgeisler/pygob#6]
Is it possible to realize the encode function of GoStrucy?

**Comment by leewz** [comment]
I started working on that [in my branch](https://github.com/leewz/pygob/commits/leewz-dump-compound), but I got stuck when it came to nested types.

For example
1. You have a custom type X.
2. And a custom type Y which contains an X.
3. You dump an object of type Y.

My dumper will detect the need to dump a description of Y, but it won't detect that X has also been dumped, so it won't dump a description of X.

As a workaround, you could manually encode the types into the stream, or, if modifying an existing gob, save and reuse the type headers from the original gob.

**Comment by andrewelkin** [comment]
I agree. Supporting things like named tuples it would be much more useful..
